.github/workflows/release.yml

```diff
@@ -0,0 +1,142 @@
+name: Release
+on:
+  push:
+    tags:
+      - '*'
+  workflow_dispatch:
+    inputs:
+      version:
+        description: 'Release version to build and upload (e.g. "v9.8.7")'
+        required: true
+      buildonly:
+        description: 'Build only: Do not create release'
+        default: "true"  # 'choice' type requires string value
+        type: choice
+        options:
+          - "true"  # Must be quoted string, boolean value not supported.
+          - "false"
+
+jobs:
+  check:
+    name: Check
+    runs-on: ubuntu-latest
+    steps:
+    - name: Determine Version
+      id: getversion
+      run: |
+        if [[ -z "${{ inputs.version }}" ]]
+        then
+              VERSION=${{ github.ref_name }}
+        else
+              VERSION=${{ inputs.version }}
+        fi
+        if ! grep -Eq 'v[0-9]+(\.[0-9]+(\.[0-9]+(-.+)?)?)?$' <<<"$VERSION"
+          then
+            echo "Unable to parse release version '$VERSION' from github event JSON, or workflow 'version' input."
+            exit 1
+          fi
+
+        echo "version=$VERSION" >> $GITHUB_OUTPUT
+        echo "::notice::Building $VERSION"
+    - name: Determine release
+      id: buildonly
+      run: |
+        # The 'tag' trigger will not have a 'buildonly' input set. Handle
+        # this case in a readable/maintainable way.
+        if [[ -z "${{ inputs.buildonly }}" ]]
+        then
+          BUILDONLY=false
+        else
+          BUILDONLY=${{ inputs.buildonly }}
+        fi
+        echo "buildonly=$BUILDONLY" >> $GITHUB_OUTPUT
+        echo "::notice::This will be build-only: $BUILDONLY"
+    outputs:
+      version: ${{ steps.getversion.outputs.version }}
+      buildonly: ${{ steps.buildonly.outputs.buildonly }}
+
+  build-artifacts:
+    name: Build Artifacts
+    runs-on: ubuntu-latest
+    needs: check
+    steps:
+    - name: Checkout Version
+      uses: actions/checkout@v4
+      with:
+        ref: ${{needs.check.outputs.version}}
+    - name: Install Protoc
+      uses: arduino/setup-protoc@v3
```

Comment:
I would prefer a normal apt-get install here, from contrib/container_images/Dockerfile.Rust
`apt-get update && apt-get -y install protobuf-compiler libprotobuf-dev`

Also do we know which rust version is used, if it is the normal ubuntu one that might be very out of date and fail to compile. Or do the github actions images always include newer versions?

(@baude @mheon Why do we even build a binary that we upload? It is not a static one so it is unlikely to run everywhere, IMO just the vendor tarball is all we need, but anyway that is besides the release action here so I am fine to defer that discussion to another day)

Reply:
Rust is by default installed in the GHA, I'm not actually sure of the version, I can check it. I'm not too familiar with rust, but I can just run `rustup update` in the action to make sure we're always on latest, if that sounds like a good idea?